middleware/response.go

```diff
@@ -67,15 +88,27 @@ func (b *badResponseLoggingWriter) WriteHeader(statusCode int) {
 }
 
 // Hijack hijacks the first response writer that is a Hijacker.
-func (b *badResponseLoggingWriter) Hijack() (net.Conn, *bufio.ReadWriter, error) {
+func (b *nonFlushingBadResponseLoggingWriter) Hijack() (net.Conn, *bufio.ReadWriter, error) {
 	hj, ok := b.rw.(http.Hijacker)
 	if ok {
 		return hj.Hijack()
 	}
 	return nil, nil, fmt.Errorf("badResponseLoggingWriter: can't cast underlying response writer to Hijacker")
 }
 
-func (b *badResponseLoggingWriter) captureResponseBody(data []byte) {
+func (b *nonFlushingBadResponseLoggingWriter) getStatusCode() int {
+	return b.statusCode
+}
+
+func (b *nonFlushingBadResponseLoggingWriter) getWriteError() error {
+	return b.writeError
+}
+
+func (b *flushingBadResponseLoggingWriter) Flush() {
+	b.f.Flush()
+}
```

Comment:
[super nit] I would move this below, to keep all `nonFlushingBadResponseLoggingWriter` receiver functions grouped together.